packages/mobile/src/components/offline-downloads/DownloadToggle.tsx

```diff
@@ -110,21 +112,45 @@ export const DownloadToggle = ({
   const isCollectionMarkedForDownload = useSelector(
     getIsCollectionMarkedForDownload(collectionIdStr)
   )
+
   const handleToggleDownload = useCallback(
     (isDownloadEnabled: boolean) => {
       if (!collectionId && !isFavoritesDownload) return
       if (isDownloadEnabled) {
         downloadCollectionById(collectionId, isFavoritesDownload)
         batchDownloadTrack(tracksForDownload)
       } else {
-        collectionIdStr &&
-          removeCollectionDownload(collectionIdStr, tracksForDownload)
+        if (!isFavoritesDownload && collectionIdStr) {
+          // we are trying to remove download from a collection page
+          dispatch(
+            setVisibility({
+              drawer: 'RemoveDownloadedCollection',
+              visible: true,
+              data: { collectionId: collectionIdStr, tracksForDownload }
```

Comment:
Cool! I was expecting the handler to be passed in, but this works

Reply:
yeah im trying to avoid passing non-serializable things into redux store when possible